guava/src/com/google/common/hash/AbstractHashFunction.java

```diff
@@ -60,7 +61,7 @@ public HashCode hashBytes(byte[] input) {
   }
 
   @Override
-  public HashCode hashBytes(byte[] input, int off, int len) {
+  public HashCode hashBytes(byte[] input, @NonNegative int off, @NonNegative int len) {
```

Comment:
This method overrides `HashFunction.hashBytes`. I recommend starting by annotating the base class according to the specification. Then you can usually use the same annotations everywhere the method is overriden.
Also the `Hasher` interface and `AbstractHasher` class are good candidates for being annotated first.